pkg/solana/config/multinode.go

```diff
@@ -1,87 +1,163 @@
+	finalizedBlockPollInterval *config.Duration
+	enforceRepeatableRead      *bool
+	deathDeclarationDelay      *config.Duration
 
 	// Chain Configs
-	nodeNoNewHeadsThreshold      time.Duration
-	noNewFinalizedHeadsThreshold time.Duration
-	finalityDepth                uint32
-	finalityTagEnabled           bool
-	finalizedBlockOffset         uint32
+	nodeNoNewHeadsThreshold      *config.Duration
+	noNewFinalizedHeadsThreshold *config.Duration
+	finalityDepth                *uint32
+	finalityTagEnabled           *bool
+	finalizedBlockOffset         *uint32
 }
 
 func (c *MultiNode) MultiNodeEnabled() bool {
-	return c.multiNodeEnabled
+	return c.multiNodeEnabled != nil && *c.multiNodeEnabled
 }
 
 func (c *MultiNode) PollFailureThreshold() uint32 {
-	return c.pollFailureThreshold
+	return *c.pollFailureThreshold
 }
 
 func (c *MultiNode) PollInterval() time.Duration {
-	return c.pollInterval
+	return c.pollInterval.Duration()
 }
 
 func (c *MultiNode) SelectionMode() string {
-	return c.selectionMode
+	return *c.selectionMode
 }
 
 func (c *MultiNode) SyncThreshold() uint32 {
-	return c.syncThreshold
+	return *c.syncThreshold
 }
 
 func (c *MultiNode) NodeIsSyncingEnabled() bool {
-	return c.nodeIsSyncingEnabled
+	return *c.nodeIsSyncingEnabled
 }
 
+func (c *MultiNode) LeaseDuration() time.Duration { return c.leaseDuration.Duration() }
+
 func (c *MultiNode) FinalizedBlockPollInterval() time.Duration {
-	return c.finalizedBlockPollInterval
+	return c.finalizedBlockPollInterval.Duration()
 }
 
-func (c *MultiNode) EnforceRepeatableRead() bool {
-	return c.enforceRepeatableRead
-}
+func (c *MultiNode) EnforceRepeatableRead() bool { return *c.enforceRepeatableRead }
 
-func (c *MultiNode) DeathDeclarationDelay() time.Duration {
-	return c.deathDeclarationDelay
-}
+func (c *MultiNode) DeathDeclarationDelay() time.Duration { return c.deathDeclarationDelay.Duration() }
 
 func (c *MultiNode) NodeNoNewHeadsThreshold() time.Duration {
-	return c.nodeNoNewHeadsThreshold
+	return c.nodeNoNewHeadsThreshold.Duration()
 }
 
 func (c *MultiNode) NoNewFinalizedHeadsThreshold() time.Duration {
-	return c.noNewFinalizedHeadsThreshold
+	return c.noNewFinalizedHeadsThreshold.Duration()
 }
 
-func (c *MultiNode) FinalityDepth() uint32 {
-	return c.finalityDepth
-}
+func (c *MultiNode) FinalityDepth() uint32 { return *c.finalityDepth }
 
-func (c *MultiNode) FinalityTagEnabled() bool {
-	return c.finalityTagEnabled
-}
+func (c *MultiNode) FinalityTagEnabled() bool { return *c.finalityTagEnabled }
 
-func (c *MultiNode) FinalizedBlockOffset() uint32 {
-	return c.finalizedBlockOffset
-}
+func (c *MultiNode) FinalizedBlockOffset() uint32 { return *c.finalizedBlockOffset }
 
 func (c *MultiNode) SetDefaults() {
-	// TODO: Set defaults for MultiNode config https://smartcontract-it.atlassian.net/browse/BCI-4065
-	c.multiNodeEnabled = false
+	c.multiNodeEnabled = ptr(false)
+
+	// Node Configs
+	c.pollFailureThreshold = ptr(uint32(5))
```

Comment:
The default values here - where are they coming from? For example, the 5 for syncThreshold or the 10 for pollInterval. Could these come from another source or if they are to be hardcoded, could we provide some documentation as to why these are the best default values for these fields?

Reply:
I has a meeting with Dmytro and Aaron where we discussed what default values make sense for each config. I can for sure add some comments though!